optuna/testing/storages.py

```diff
@@ -44,8 +44,13 @@
 
 
 class StorageSupplier:
-    def __init__(self, storage_specifier: str, **kwargs: Any) -> None:
+    def __init__(
+        self, storage_specifier: str, base_storage: BaseStorage | None = None, **kwargs: Any
+    ) -> None:
         self.storage_specifier = storage_specifier
+        if base_storage is not None:
+            assert storage_specifier == "grpc_proxy"
```

Comment:
My initial impression is that it could potentially be a bit confusing that `storage_specifiers` accepts "grpc_proxy", even though "grpc_rdb" and "grpc_journal_file" still exist. I have no idea how to mitigate this right now. Therefore, I'll comment later with a suggested solution.

Reply:
I've been thinking about it, but I could not come up with a simple solution.

It seems that larger changes might be needed, which should not be addressed in this PR. Considering that this bug fix should be included in the next minor release, I will merge this PR first and then considering whether to submit a follow-up PR.

I apologize for the delayed review. I will proceed with merging this.